### ISSUE TITLE
Startup needs to be logged when running as service.

### ISSUE THREAD

**Issue by pwhodges** [mholt/caddy#1205]
When starting Caddy at the command line, it is clear what is happening, and in particular the sites being served are listed out.  When Caddy is running as a service this information is lost, because only stderr is captured in the log file. (Tested on Windows; but I presume *nix will be the same.)

Having the initial console messages up to and including the list of domains in the log would make it much easier to see when Caddy has been restarted, which would be be useful in assessing the errors before and after. It might even be useful to precede it in the log by a standard message saying "Caddy started" to make scripts which search in the log a little easier.

**Comment by tobya** [comment]
@elcore @mholt 

in[ server.go OnStartupComplete the call to fmt.Println](https://github.com/mholt/caddy/blob/master/caddyhttp/httpserver/server.go#L365) outputs the servername being served.

However replacing

    		fmt.Println(output)
with 

		log.Println(output)

will cause a number of issues

1. list of hosts will only be show when -log is used on command line
1. also the date is printed infront of the host name `2016/10/23 09:15:34 http://127.0.0.1:1234` which is not a problem but is less asthetically pleasing

Is there a simple solution to these issues with log ?  I did wonder about keepting fmt.Println and building a list of hosts to be output when all servers have been listed and then outputing via log.Println but I would need to store the string list on a global level as OnStartupComplete is called (and logged) for each host.

Any pointers of how this should be implemented?

**Comment by elcore** [comment]
Hello @tobya,

In Linux (e.g. Ubuntu/Debian) one can use `sudo systemctl status caddy`

```
Oct 22 10:10:10 host systemd[1]: Started Caddy HTTP/2 web server .
Oct 22 10:10:10 host caddy[3314]: Activating privacy features... done.
Oct 22 10:10:10 host caddy[3314]: https://domain.tld
Oct 22 10:10:10 host caddy[3314]: http://domain.tld
```

You could use something like (non-tested):

```
// OnStartupComplete lists the sites served by this server
// and any relevant information, assuming caddy.Quiet == false.
func (s *Server) OnStartupComplete() {
        if caddy.Quiet {
                return
        }

        for _, site := range s.sites {
                output := site.Addr.String()
                if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
                        output += " (only accessible on this machine)"
                }
                if runtime.GOOS == "windows" {
                        l := log.New(os.Stdout, "[Caddy] Serving: ", 0)
                        l.Println(output)
                } else {
                        fmt.Println(output)
                }
        }
}
```

**Comment by mholt** [comment]
Logs are going to have the timestamp at the beginning. I think instead of replacing fmt.Println with log.Println, we'll just add log.Println after fmt.Println and that should do the trick.

**Comment by elcore** [comment]
> Logs are going to have the timestamp at the beginning.

log.SetFlags(0) should remove the timestamp

> I think instead of replacing fmt.Println with log.Println, we'll just add log.Println after fmt.Println and that should do the trick.

I would do this

```
for _, site := range s.sites {
	output := site.Addr.String()
	if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
		output += " (only accessible on this machine)"
	}
	if runtime.GOOS == "windows" {
		log.SetFlags(0)
		log.SetPrefix("[Caddy] Serving: ")
		log.Println(output)
	} else {
		fmt.Println(output)
	}
}
```

**Comment by mholt** [comment]
I don't think we should remove the timestamps from the logs. And let's just keep it simple, same thing on any OS.

**Comment by elcore** [comment]
Okay @mholt 😄 -- So, who of us needs a Hacktoberfest T-Shirt 😉?

**Comment by tobya** [comment]
@mholt Ok, I'll keep it simple and simply add the additional call to log.Println

**Comment by tobya** [comment]
Sorry about all the untidy commits, the last one 	4cc6b29 is the only one I actually submitted in this pull request.

**Comment by mholt** [comment]
Implemented in #1211 - thanks!
